### DISCUSSION TITLE
A way to inject database connection in websocket

### DISCUSSION THREAD

**Discussion by TheAwiteb** [salvo-rs/salvo#659] [Q&A] [answered]
Hello, I want to use the database in websocket, but it seems to me that this [is not possible](https://github.com/salvo-rs/salvo/blob/dfbe2b0a4768f0ce35fd07c561f11322aa0427eb/crates/extra/src/websocket.rs#L131). Is there a way other than making the database static?

**Comment by chrislearn** [comment, marked as answer]
How about use closure? just like this example move user into closure?

https://github.com/salvo-rs/salvo/blob/89b36ab2a53c13e7b72cd146f929ff4c007fae6c/examples/websocket/src/main.rs#L11C1-L31C2

**Reply by TheAwiteb** [reply]
Nice, thank you
